agenda/presentation/src/main/java/com/nibalk/tasky/agenda/presentation/detail/DetailScreen.kt

```diff
@@ -101,7 +113,39 @@ fun DetailScreen(
     state: DetailState,
     onAction: (DetailAction) -> Unit,
 ) {
+    val context = LocalContext.current
+    val itemModifier: Modifier = Modifier.padding(
+        horizontal = MaterialTheme.spacing.spaceMedium
+    )
+
+    val compressedPhotos = remember { mutableStateListOf<ByteArray?>() }
+    val compressedPhotosTemp = remember { mutableStateListOf<ByteArray?>() }
+
+    if (state.agendaType == AgendaType.EVENT) {
+        LaunchedEffect(state.details.asEventDetails.photos) {
+            compressedPhotosTemp.clear()
+            state.details.asEventDetails.photos.forEach { eventPhoto ->
+                val compressedByteArray = when(eventPhoto) {
+                    is EventPhoto.Local -> {
+                        withContext(Dispatchers.IO) {
+                            Uri.parse(eventPhoto.location).getCompressedByteArray(context)
+                        }
```

Comment:
If `getCompressedByteArray` is a suspend function, the responsibility to switch to IO lies at that function instead of the caller. Otherwise, the caller of it will always have to remember switching to the right dispatcher